cmd/state-installer/internal/installer/installation.go

```diff
@@ -1,125 +1,69 @@
-		}
+func (i *Installation) Install() error {
+	if err := i.PrepareBinTargets(); err != nil {
+		return errs.Wrap(err, "Could not prepare for installation")
+	}
+	if err := fileutils.MkdirUnlessExists(i.binaryDir); err != nil {
+		return errs.Wrap(err, "Could not create target directory: %s", i.binaryDir)
 	}
-	if len(es) > 0 {
-		return errs.Wrap(es[0], "Failed to remove some back-up files")
+	if err := fileutils.CopyAndRenameFiles(filepath.Join(i.fromDir, "bin"), i.binaryDir); err != nil {
+		return errs.Wrap(err, "Failed to copy installation files to dir %s", i.binaryDir)
+	}
+	if err := InstallSystemFiles(filepath.Join(i.fromDir, "system"), i.binaryDir, i.appDir); err != nil {
+		return errs.Wrap(err, "Installation of system files failed.")
 	}
 
 	return nil
 }
 
-func (i *Installation) BackupFiles() error {
-	// Todo: https://www.pivotaltracker.com/story/show/177600107
-	// Get target file paths.
-	var targetFiles []string
-	for _, file := range fileutils.ListDir(i.fromDir, false) {
-		targetFile := filepath.Join(i.binaryDir, filepath.Base(file))
-		targetFiles = append(targetFiles, targetFile)
+// PrepareBinTargets will move aside any targets in the bin dir that we would otherwise overwrite.
+// This guards us from file in use errors as well as false positives by security software
+func (i *Installation) PrepareBinTargets() error {
+	files, err := ioutil.ReadDir(filepath.Join(i.fromDir, "bin"))
+	if err != nil {
+		return errs.Wrap(err, "Could not read target dir")
 	}
-	logging.Debug("Target files=%s", strings.Join(targetFiles, ","))
 
-	backups, err := backupFiles(targetFiles)
+	temp, err := ioutil.TempDir("", "update-from-state-"+constants.Version)
 	if err != nil {
-		return errs.Wrap(err, "Backup of existing files failed.")
+		return errs.Wrap(err, "Could not access temp dir")
 	}
-	i.backups = append(i.backups, backups...)
-	return nil
-}
 
-func (i *Installation) Rollback() error {
-	return restoreFiles(i.backups)
-}
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
 
-func (i *Installation) Install() error {
-	if err := i.BackupFiles(); err != nil {
-		return errs.Wrap(err, "Failed to backup original files.")
-	}
-	if err := fileutils.MkdirUnlessExists(i.binaryDir); err != nil {
-		return errs.Wrap(err, "Could not create target directory: %s", i.binaryDir)
-	}
-	if err := fileutils.CopyAndRenameFiles(filepath.Join(i.fromDir, "bin"), i.binaryDir); err != nil {
-		return errs.Wrap(err, "Failed to copy installation files to dir %s", i.binaryDir)
-	}
-	if err := InstallSystemFiles(filepath.Join(i.fromDir, "system"), i.binaryDir, i.appDir); err != nil {
-		return errs.Wrap(err, "Installation of system files failed.")
+		targetFile := filepath.Join(i.binaryDir, file.Name())
+		if fileutils.TargetExists(targetFile) {
+			if err := os.Rename(targetFile, filepath.Join(temp, file.Name())); err != nil {
+				return errs.Wrap(err, "Could not move executable aside prior to install: %s", targetFile)
+			}
```

Comment:
Unfortunately, this might be a problem.  Because `os.Rename` does not move across disks.  So, if the State Tool installation is on a different disk than the `/tmp` directory, it will fail with a weird error: https://github.com/golang/go/issues/13766#issuecomment-167941305

Reply:
Thanks for spotting that. I've moved it to be a Windows only mechanic and targeted the same folder as the original (like you had originally done). I also added some cleanup code but it's only triggered on a new install. Meaning on Windows you'll always have the old variant around as well, but since it doesn't have an executable extension name it's no runnable so it's only using up disk space.

I figure we can improve this once we can depend more on the state-svc. For now I think it makes sense.